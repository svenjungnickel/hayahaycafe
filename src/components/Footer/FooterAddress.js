import React from 'react';
import PropTypes from 'prop-types';
import { graphql, StaticQuery } from 'gatsby';

export const ListFooterAddress = ({ address }) => {
    const filteredAddressItems = Object.values(address).filter((item) => '' !== item);

    return <p>{0 < filteredAddressItems.length && <>{filteredAddressItems.join(', ')}</>}</p>;
};

ListFooterAddress.propTypes = {
    address: PropTypes.shape({
        company: PropTypes.string,
        addressLine1: PropTypes.string,
        addressLine2: PropTypes.string,
        province: PropTypes.string,
        postalCode: PropTypes.string,
        country: PropTypes.string,
    }).isRequired,
};

const FooterAddress = () => (
    <StaticQuery
        query={graphql`
            query {
                globalSettings: settingsYaml {
                    location {
                        address {
                            company
                            addressLine1
                            addressLine2
                            province
                            postalCode
                            country
                        }
                    }
                }
            }
        `}
        render={({ globalSettings }) => (
            <>
                {!!globalSettings && !!globalSettings.location && !!globalSettings.location.address && (
                    <ListFooterAddress address={globalSettings.location.address} />
                )}
            </>
        )}
    />
);

export default FooterAddress;
