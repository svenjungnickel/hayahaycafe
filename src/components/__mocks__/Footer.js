import React from 'react';

const mockFooter = () => <footer />;

export default mockFooter;
