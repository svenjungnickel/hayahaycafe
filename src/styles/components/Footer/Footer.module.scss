.footer {
    color: white;
    margin-top: 70px;

    .footer__pre {
        padding: 20px 0;
        background-color: var(--footerPre-bg);
    }

    .footer__main {
        padding: 40px 0;
        background-color: var(--footerMain-bg);

        a {
            color: white;

            &.active,
            &:focus,
            &:hover {
                color: var(--color-secondary);
            }
        }

        .footer__address {
            margin-top: 10px;
        }
    }
}
