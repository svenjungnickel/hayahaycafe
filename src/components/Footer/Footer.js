import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import { Container, Nav } from 'react-bootstrap';
import SocialMediaIcons from './SocialMediaIcons';
import FooterAddress from './FooterAddress';
import FooterStyles from '../../styles/components/Footer/Footer.module.scss';

export const NavItem = ({ link, name }) => (
    <Nav.Item>
        <Link to={link} className="nav-link">
            {name}
        </Link>
    </Nav.Item>
);

NavItem.propTypes = {
    link: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
};

const Footer = () => (
    <footer className={FooterStyles.footer}>
        <div className={FooterStyles.footer__pre}>
            <SocialMediaIcons />
        </div>
        <div className={FooterStyles.footer__main}>
            <Container className="text-center">
                <Nav className="justify-content-center">
                    <NavItem link="/contact" name="Contact" />
                    <NavItem link="/legal" name="Legal" />
                    <NavItem link="/data-privacy" name="Data Privacy" />
                </Nav>
                <div className={FooterStyles.footer__address}>
                    <FooterAddress />
                </div>
            </Container>
        </div>
    </footer>
);

export default Footer;
