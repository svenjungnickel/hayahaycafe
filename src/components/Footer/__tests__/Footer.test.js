import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../Footer';

jest.mock('../SocialMediaIcons');
jest.mock('../FooterAddress');

describe('Footer', () => {
    it('renders correctly', () => {
        const component = <Footer />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
