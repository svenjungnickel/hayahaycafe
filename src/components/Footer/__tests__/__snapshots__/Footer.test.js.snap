// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="footer__pre"
  >
    <ul>
      <li>
        facebook
      </li>
      <li>
        instagram
      </li>
      <li>
        tripadvisor
      </li>
    </ul>
  </div>
  <div
    className="footer__main"
  >
    <div
      className="text-center container"
    >
      <div
        className="justify-content-center nav"
        onKeyDown={[Function]}
      >
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/contact"
          >
            Contact
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/legal"
          >
            Legal
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/data-privacy"
          >
            Data Privacy
          </a>
        </div>
      </div>
      <div
        className="footer__address"
      >
        <p>
          FooterAddress
        </p>
      </div>
    </div>
  </div>
</footer>
`;
